### ISSUE TITLE
new runtime check fails for docker build

### ISSUE THREAD

**Issue by stevie-** [terraform-aws-modules/terraform-aws-lambda#361]
## Description

With the latest version [4.1.2 and 4.1.3](https://github.com/terraform-aws-modules/terraform-aws-lambda/commit/dfc8934e907e5eb7f1820b838ec6e98f4011128a#diff-34b2a63c4b18a635042766329eca28d8801699eecf5e93eaedaeda1309931381L655-L663) a new shell check for `runtime` parameter was introduced. This breaks our workflows on github runner ubuntu-20.04 (or any runner which has not the selected runtime version installed). We still set `python3.7` runtime which is not available on the runner, which is known and thus we use docker to build our lambdas.

It seems that this check is executed before the build process switches to the docker runtime.
Please check why this code is not working with the docker build setting.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 4.1.3

- Terraform version:
v1.2.9
- Provider version(s):
hashicorp/local v2.2.3
hashicorp/aws v4.36.0
hashicorp/null v3.1.1
hashicorp/external v2.2.2

## Reproduction Code [Required]
```hcl
{
  create      = true
  create_role = true

  function_name = "python3.7-docker-build-fails"
  handler       = "app.lambda_handler"
  runtime       = "python3.7"
  source_path = [
    {
      path             = "../lambdas/trigger-lambda/src",
      pip_requirements = "../lambdas/trigger-lambda/requirements.txt",
    }
  ]

  build_in_docker   = true
  docker_image      = "lambci/lambda:build-python3.7"
  docker_build_root = "../lambdas/trigger-lambda/docker"
}
```

```python
# requirements.txt
jsonpath-ng==1.5.2
```

< Are you using workspaces? > no workspaces
< Have you cleared the local cache (see Notice section above)? > clean cache
< List steps in order that led up to the issue you encountered > update module to 4.1.3 and rerun unchanged code

## Expected behavior

The check for runtime binary should be done inside docker container.

## Actual behavior

```log
│ Error: External Program Execution Failed
│
│   with module.lambda.data.external.archive_prepare[0],
│   on .terraform/modules/lambda/package.tf line 10, in data "external" "archive_prepare":
│   10:   program = [local.python, "${path.module}/package.py", "prepare"]
│
│ The data source received an unexpected error while attempting to execute
│ the program.
│
│ Program: /path/to/bin/python3
│ Error Message: Traceback (most recent call last):
│   File "REDACTED/aws/lambda_function/.terraform/modules/lambda/package.py", line 1403, in <module>
│     main()
│   File "REDACTED/aws/lambda_function/.terraform/modules/lambda/package.py", line 1399, in main
│     exit(args.command(args))
│   File "REDACTED/aws/lambda_function/.terraform/modules/lambda/package.py", line 1198, in prepare_command
│     build_plan = bpm.plan(source_path, query)
│   File "REDACTED/aws/lambda_function/.terraform/modules/lambda/package.py", line 768, in plan
│     pip_requirements_step(pip_requirements, prefix,
│   File "REDACTED/aws/lambda_function/.terraform/modules/lambda/package.py", line 661, in pip_requirements_step
│     raise RuntimeError(
│ RuntimeError: Python interpreter version equal to defined lambda runtime
│ (python3.7) should be available in system PATH
│
│ State: exit status
```

**Comment by lorengordon** [comment]
Related question: What is the appetite for running docker during the plan phase? Obviously you would need docker, and it would need to pull the image...

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.1.4 :tada:
